crates/bevy_debug_draw/src/debug_draw.rs

```diff
@@ -0,0 +1,200 @@
+        let trf = (position + rotation * vec3(half_size.x, half_size.y, half_size.z)).to_array();
+        let blf = (position + rotation * vec3(-half_size.x, -half_size.y, half_size.z)).to_array();
+        let brf = (position + rotation * vec3(half_size.x, -half_size.y, half_size.z)).to_array();
+        // Back
+        let tlb = (position + rotation * vec3(-half_size.x, half_size.y, -half_size.z)).to_array();
+        let trb = (position + rotation * vec3(half_size.x, half_size.y, -half_size.z)).to_array();
+        let blb = (position + rotation * vec3(-half_size.x, -half_size.y, -half_size.z)).to_array();
+        let brb = (position + rotation * vec3(half_size.x, -half_size.y, -half_size.z)).to_array();
+        self.positions.extend([
+            tlf, trf, trf, brf, brf, blf, blf, tlf, // Front
+            tlb, trb, trb, brb, brb, blb, blb, tlb, // Back
+            tlf, tlb, trf, trb, brf, brb, blf, blb, // Front to back
+        ]);
+        self.colors
+            .extend(std::iter::repeat(color.as_linear_rgba_f32()).take(24));
+    }
+
+    /// Draw a line from `start` to `end`.
+    #[inline]
+    pub fn line_2d(&mut self, start: Vec2, end: Vec2, color: Color) {
+        self.line_gradient_2d(start, end, color, color);
+    }
+
+    /// Draw a line from `start` to `end`.
+    #[inline]
+    pub fn line_gradient_2d(
+        &mut self,
+        start: Vec2,
+        end: Vec2,
+        start_color: Color,
+        end_color: Color,
+    ) {
+        self.line_gradient(start.extend(0.), end.extend(0.), start_color, end_color);
+    }
+
+    /// Draw a line from `start` to `start + vector`.
+    #[inline]
+    pub fn ray_2d(&mut self, start: Vec2, vector: Vec2, color: Color) {
+        self.ray_gradient_2d(start, vector, color, color);
+    }
+
+    /// Draw a line from `start` to `start + vector`.
+    #[inline]
+    pub fn ray_gradient_2d(
+        &mut self,
+        start: Vec2,
+        vector: Vec2,
+        start_color: Color,
+        end_color: Color,
+    ) {
+        self.line_gradient_2d(start, start + vector, start_color, end_color);
+    }
+
+    // Draw a circle.
+    #[inline]
+    pub fn circle_2d(&mut self, position: Vec2, radius: f32, color: Color) {
+        self.circle(position.extend(0.), Vec3::Z, radius, color);
+    }
+
+    /// Draw a rectangle.
+    #[inline]
+    pub fn rect_2d(&mut self, position: Vec2, rotation: f32, size: Vec2, color: Color) {
+        self.rect(
+            position.extend(0.),
+            Quat::from_rotation_z(rotation),
+            size,
+            color,
+        );
+    }
+
+    /// Clear everything drawn up to this point, this frame.
+    #[inline]
+    pub fn clear(&mut self) {
+        self.positions.clear();
+        self.colors.clear();
+    }
+
+    /// Take the positions and colors data from `self` and overwrite the `mesh`'s vertex positions and colors.
+    #[inline]
+    pub fn update_mesh(&mut self, mesh: &mut Mesh) {
```

Comment:
Does this need to be `pub`? Can it just be `pub(crate)`?

Alternatively, could we embed the Mesh here so it's edited in place instead of batch copied? This would avoid the extra copy.

Reply:
It doesn't seem like anything is getting copied?